tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java

```diff
@@ -0,0 +1,740 @@
+                .withRequiredArg()
+                .describedAs("timestamp/-1(latest)/-2(earliest)")
+                .ofType(Long.class)
+                .defaultsTo(-1L);
+            reportIntervalOpt = parser.accepts("report-interval-ms", "The reporting interval.")
+                .withRequiredArg()
+                .describedAs("ms")
+                .ofType(Long.class)
+                .defaultsTo(30_000L);
+            options = parser.parse(args);
+            if (args.length == 0 || options.has(helpOpt)) {
+                CommandLineUtils.printUsageAndExit(parser, "Validate that all replicas for a set of topics have the same data.");
+            }
+            if (options.has(versionOpt)) {
+                CommandLineUtils.printVersionAndExit();
+            }
+            CommandLineUtils.checkInvalidArgs(parser, options, bootstrapServerOpt, brokerListOpt);
+            CommandLineUtils.checkInvalidArgs(parser, options, topicsIncludeOpt, topicWhiteListOpt);
+            if (!options.has(bootstrapServerOpt) && !options.has(brokerListOpt)) {
+                CommandLineUtils.printUsageAndExit(parser, format("The %s option is required", bootstrapServerOpt));
+            }
+            if (options.has(brokerListOpt)) {
+                System.out.printf("WARNING: The %s option is deprecated and will be removed. " +
+                    "Use the %s option with the same syntax.%n", brokerListOpt, bootstrapServerOpt);
+            }
+            if (options.has(topicWhiteListOpt)) {
+                System.out.printf("WARNING: The %s option is deprecated and will be removed. " +
+                    "Use the %s option with the same syntax.%n", topicWhiteListOpt, topicsIncludeOpt);
+            }
+        }
+
+        String brokerHostsAndPorts() {
+            String brokerList = options.valueOf(options.has(bootstrapServerOpt) ? bootstrapServerOpt : brokerListOpt);
+            ToolsUtils.validateBootstrapServer(parser, brokerList);
+            return brokerList;
+        }
+
+        TopicFilter.IncludeList topicsIncludeFilter() {
+            String regex = options.valueOf(options.has(topicsIncludeOpt) ? topicsIncludeOpt : topicWhiteListOpt);
+            try {
+                Pattern.compile(regex);
+            } catch (PatternSyntaxException e) {
+                throw new RuntimeException(format("%s is an invalid regex", regex));
+            }
+            return new TopicFilter.IncludeList(regex);
+        }
+
+        int fetchSize() {
+            return options.valueOf(fetchSizeOpt);
+        }
+
+        int maxWaitMs() {
+            return options.valueOf(maxWaitMsOpt);
+        }
+
+        long initialOffsetTime() {
+            return options.valueOf(initialOffsetTimeOpt);
+        }
+
+        long reportInterval() {
+            return options.valueOf(reportIntervalOpt);
+        }
+    }
+
+    private static class MessageInfo {
+        private final int replicaId;
```

Comment:
I don't think we need these fields to be `private`

Reply:
Well no. Thanks.